052-cruise-control-api-users.md

```diff
@@ -0,0 +1,269 @@
+# Add ability to create Cruise Control REST API users
+
+This proposal is about giving developers the ability to create REST API users for the Cruise Control REST API.
+This would allow roles and permissions to be defined to allow developers and third-party applications to access the Cruise Control REST API without having to disable HTTP basic authentication.
+
+## Current situation
+
+At this time, Strimzi creates two Cruise Control API users:
+
+* `admin`      - Used by Strimzi for Cruise Control admin operations.
+* `heathcheck` - Used by Strimzi for checking the readiness of the Cruise Control application.
+
+Access to these API users is limited to Strimzi.
+So when HTTP basic authentication is enabled for Cruise Control, no one can access the Cruise Control API other than Strimzi.
+
+### How API user management is handled elsewhere
+
+Cruise Control is not the only Kafka component which has API users.
+Although this proposal is stictly about Cruise Control API users, for consistency across the code base, we also considered the following components when designing our solution:
+
+* Kafka users
+* KafkaConnect users
+
+#### Kafka users
+
+In Strimzi, Kafka users are managed via custom resources.
+
+```yaml=
+apiVersion: kafka.strimzi.io/v1beta2
+kind: KafkaUser
+metadata:
+  name: my-user-1
+  labels:
+    strimzi.io/cluster: my-cluster
+spec:
+  authentication:
+    type: tls
+  authorization:
+    type: simple
+    acls:
+      ...
+```
+These resources are watched and managed by the the Strimzi User Operator, which creates Kafka users based on the descriptions provided by the resources.
+
+For more details, checkout the Strimzi documentation [here](https://strimzi.io/docs/operators/latest/deploying#proc-configuring-kafka-user-str)
+
+#### KafkaConnect users
+
+At this time, Strimzi doesn't offer any method of creating or configuring KafkaConnect API users.
+That being said, the direction we take with creating and managing Cruise Control API users could be used as a precedent for its design in the future.
+
+## Motivation
+
+There are a couple use cases where developers would want to access the Cruise Control API without disabling API security:
+
+* Monitoring a Strimzi-managed Kafka cluster with the Cruise Control UI.
+* Gathering Cruise Control specific statistical information that is not currently available via Strimzi Operator or Cruise Control sensor metrics e.g. detailed information surrounding cluster and partition load and user tasks.
+* Experimenting with Cruise Control operations not currently supported by Strimzi.
```

Comment:
Originally was referencing POST operations before we decided to strip them out for this proposal. Going to remove this line now since we are not supporting ADMIN operations